Lib/compileall.py

```diff
@@ -157,8 +157,8 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
         dfile = os.path.join(ddir, name)
 
     if stripdir is not None:
-        fullname_parts = fullname.split(os.path.sep)
-        stripdir_parts = stripdir.split(os.path.sep)
+        fullname_parts = os.fspath(fullname).split(os.path.sep)
```

Comment:
This makes the logic on line 151 redundant. Not sure why that's gated by the condition on `quiet`, though.

Reply:
Oh, yes. It is odd.

@berkerpeksag sorry for bothering, do you remember why the quiet check is there? It looks like `quiet` is being used across the code to provide some set some kind of failing level here. The docstring is also a bit ambiguous.

> full output with False or 0, errors only with 1, no output with 2

I am not sure how to parse the behavior for `1`.

Should this behavior be changed? If not, the docstring should definitely be touched up.

Reply:
Anyway, I am moving the path-like object translation up there.

Reply:
I also added the other path arguments.